### ISSUE TITLE
setting Border to False, with items in a graphics layout is problematic...

### ISSUE THREAD

**Issue by j9ac9k** [pyqtgraph/pyqtgraph#1593]
### Short description

pg.GraphicsLayoutWidget(border=False) raises a TypeError, while `border=True` is accepted

### Code to reproduce

```python
import sys
import pyqtgraph as pg
from pyqtgraph.Qt import QtWidgets, QtGui
import numpy as np

app = QtWidgets.QApplication(sys.argv)
w = pg.GraphicsLayoutWidget(border=False)
plot = w.addPlot(y=np.random.normal(size=100))
w.show()

if __name__ == "__main__":
    QtGui.QApplication.instance().exec_()
```

### Expected behavior
No border is shown

### Real behavior

```
Traceback (most recent call last):
  File "problem.py", line 9, in <module>
    plot = w.addPlot(y=np.random.normal(size=100))
  File "/Users/ogi/Developer/pyqtgraph/pyqtgraph/graphicsItems/GraphicsLayout.py", line 71, in addPlot
    self.addItem(plot, row, col, rowspan, colspan)
  File "/Users/ogi/Developer/pyqtgraph/pyqtgraph/graphicsItems/GraphicsLayout.py", line 130, in addItem
    borderRect.setPen(fn.mkPen(self.border))
  File "/Users/ogi/Developer/pyqtgraph/pyqtgraph/functions.py", line 356, in mkPen
    color = mkColor(color)
  File "/Users/ogi/Developer/pyqtgraph/pyqtgraph/functions.py", line 281, in mkColor
    raise TypeError(err)
TypeError: Not sure how to make a color from "(False,)"
```

### Tested environment(s)

 * PyQtGraph version: master: d9eb8e1afbffd0aa12ee22fc5fc329dcd04d16b1
 * Qt Python binding: PyQt5 5.12.3
 * Python version:  python 3.7.9
 * NumPy version: 1.20.1
 * Operating system: macOS
 * Installation method: pip

### Additional context

This is an issue with an API type incompatibility, where `True` is recognized, but `False` isn't, where `False` should be treated equivalent to None by `setPen`.

**Comment by j9ac9k** [comment]
I should also add that this is the sort of thing that type-hinting will _really_ help with...

**Comment by j9ac9k** [comment]
@jennifer-manriquez please do!

**Comment by jennifer-manriquez** [comment]
Hi @j9ac9k, I see that True is supported because it is handled in the constructor. If I understand right, to support False, maybe could be similar, and treat it like none. Like this: https://github.com/jennifer-manriquez/pyqtgraph/commit/2899143e84c51cca7baca7d405642036b839a496 This passes the pytest tests. Please let me know if this solves the issue, and I could start the PR or if not, how I could get on the right track. Also, if there is a need to write a test for this case I would be happy to do it.

**Comment by j9ac9k** [comment]
I think that diff is perfectly acceptable, and I would accept a PR with it (would probably give other maintainers an opportunity to chime in as well).

**Comment by j9ac9k** [comment]
Fixed by #1840
